tests/cpgrid/LgrChecks.hpp

```diff
@@ -168,61 +216,94 @@ void Opm::checkEqMinMaxGlobalCellLevelZeroAndLeaf(const std::vector<int>& global
+    BOOST_CHECK_EQUAL( child_level, element.level());
+    BOOST_CHECK(std::any_of(siblings_list.begin(), siblings_list.end(),
+                            [&](int sibling) { return sibling == element.index(); }));
     BOOST_CHECK_EQUAL( siblings_list.size(), expected_total_children);
 }
 
 void Opm::checkGridBasicHiearchyInfo(const Dune::CpGrid& grid,
-                                     const std::vector<std::array<int,3>>& cells_per_dim_vec)
+                                     const std::vector<std::array<int,3>>& cells_per_dim_vec,
+                                     int preAdaptMaxLevel)
 {
     const int maxLevel = grid.maxLevel(); // Leaf Grid View has index maxLevel +1
-    for (int level = 0; level <= maxLevel+1; ++level) {
+    
+    for (int level = preAdaptMaxLevel+1; level <= maxLevel+1; ++level) {
         bool isLeaf = (level == maxLevel +1);
         const auto& elements = isLeaf? Dune::elements(grid.leafGridView()) : Dune::elements(grid.levelGridView(level));
         for (const auto& element : elements) {
             BOOST_CHECK_EQUAL( element.hasFather(), static_cast<bool>(element.level())); // level>0 -> 1 (true); level=0 (false)
-            BOOST_CHECK( element.getOrigin().level() == 0);
+            BOOST_CHECK( element.getOrigin().level() <= preAdaptMaxLevel);
+
+            if (isLeaf) {
+                BOOST_CHECK( !element.mightVanish() );
+                BOOST_CHECK( grid.getMark(element) == 0); // postAdapt() has been called, therefore every element gets marked with 0
+            }
 
             auto it = element.hbegin(maxLevel);
             const auto endIt = element.hend(maxLevel);
 
-            if (element.hasFather()) {
-                // If there is no nested refinement, entity.isLeaf() and it == endIt (if dristibuted_data_ is empty).
-                BOOST_CHECK( it == endIt);
-                BOOST_CHECK( element.isLeaf() );
-
-                const auto expected_total_children = cells_per_dim_vec[element.level()-1][0]*cells_per_dim_vec[element.level()-1][1]*cells_per_dim_vec[element.level()-1][2];
-                BOOST_CHECK_CLOSE(element.geometryInFather().volume(), 1./expected_total_children, 1e-6);
-
-                checkFatherAndSiblings(element.getLevelElem(), element.level(), expected_total_children, grid);
+            if (element.hasFather() && (element.level()>preAdaptMaxLevel)) {
+                // second bool discards leaf element born in refined level grids that had not been marked for refinement in the last adapt call.
+                int subdivisionsIdx = element.level()-1-preAdaptMaxLevel;
+                const auto expected_total_children = cells_per_dim_vec[subdivisionsIdx][0]*cells_per_dim_vec[subdivisionsIdx][1]*cells_per_dim_vec[subdivisionsIdx][2];
+                BOOST_CHECK_CLOSE(element.geometryInFather().volume(), 1./expected_total_children, 1e-6);   
```

Comment:
Not part of this PR, but we seem to check a rather special case here (cell is a cube).

Reply:
It could be a wrong interpretation of what geometryInFather should return. Currently, returns the geometry in the reference element, which is indeed the unit cube.